### ISSUE TITLE
Time to process a single input entry in CLAmodel is growing linearly for large inputs.

### ISSUE THREAD

**Issue by breznak** [numenta/nupic-legacy#1359]
...the model is slowing down. 
To some extent this would be acceptable with clearing of buffers/updates..but this seems to grow linearly in #of records fed into the model.

Setup:
- average CLA (2048 cols, SP+TP, RDSEncoder, params swarmed over, fixed sized (32cells/col, 128segs/cell))
- anomaly detection task
- running the (faster) cpp version of algorithms
- (HW: i7, CLA can use only 1 core, enough RAM)

Results:
- at first: ~100 Hz processing frequency 
- ~50k records -> ~30Hz
- ~80k records -> ~15Hz

In other words, going from 100 records in 1 second to 100 records in 7s!

CC @scottpurdy Do you guys observe that in Numenta?
